src/UISystem.cpp

```diff
@@ -79,19 +79,22 @@ void UISystem::handle_load(const Event& e) {
 }
 
 void UISystem::handle_key_press(const Event& e) {
+    if(current_game_state_ != GameState::START_MENU) {
+        return;
+    }
     int key = e.get_value<int>("key", true).first;
 
     if (key == SDLK_RETURN) {
-        start_bg_.toggle();
-        logo_.toggle();
-        hit_enter_or_start_.toggle();
+        start_bg_.visible_ = false;
+        logo_.visible_ = false;
+        hit_enter_or_start_.visible_ = false;
         EventSystem::queue_event(
             Event(
                 EventType::NEW_GAME_STATE,
                 "state", GameState::IN_GAME
             )
         );
-        EventSystem::remove_event_handler(EventType::KEYPRESS_EVENT);
+        current_game_state_ = GameState::IN_GAME;
```

Comment:
GameplaySystem should be the one that queues the NEW_GAME_STATE event. UISystem should handle it. e.g., The same thing would have to happen when we pause the game. GameplaySystem sends GameState::Paused, UISystem then handles new state and show pause screen.

Sorry if I wasn't clear earlier.